client/src/pages/Budget/Budget.tsx

```diff
@@ -1,13 +1,37 @@
 import React from "react";
 import { Container } from "./Budget.styles";
 import { BudgetNavbar } from "../../components";
-import { Outlet } from "react-router-dom";
+import { Outlet, useParams } from "react-router-dom";
+import { useQuery } from "react-query";
+import { getAllBudgets } from "../../API/BudgetMethods";
+import { IBudgetData } from "../../constants";
 
 const Budget = () => {
+  // const [selectedBudget, setSelectedBudget] = useState<IBudgetData | {}>({});
+  const { budgetId } = useParams();
+
+  const fetchAllBudgets = async () => {
+    const data = await getAllBudgets();
+    return data;
+  };
+
+  const { data, status, isLoading } = useQuery("budgets", fetchAllBudgets);
```

Comment:
oooooh.  I wanna know what this is.

Reply:
So this is React Query! I'm currently learning more about it - but apparently React itself is terrible for data fetching from backend - and React Query solves this problem. Some are saying it's so good that you may not even need a global state in some instances.